src/utils/filters/strip_md.ts

```diff
@@ -17,7 +17,7 @@ export const strip_md = (str: string): string => {
 		str = str.replace(/`([^`]+)`/g, '$1');  // inline code
 		str = str.replace(/```[\s\S]*?```/g, '');  // code blocks
 		str = str.replace(/~~(.*?)~~/g, '$1');  // strikethrough
-		str = str.replace(/^[-*+] \[[x ]\] /gm, '');  // task lists
+		str = str.replace(/^[-*+] (\[[x ]\] )?/gm, '');  // task lists and list items
```

Comment:
The function doesn't handle the list-item case, which results in the markdown-stripped content still containing markdown list item markers. This causes unexpected issues when generating fragment links for entire lines of content.

So I've included this in this PR.